src/components/AdminMarkerInfo/BoxInfo/BoxInfo.jsx

```diff
@@ -8,18 +8,89 @@ import {
   Textarea,
   Text,
   Button,
+  useDisclosure,
 } from '@chakra-ui/react';
 import { ChevronLeftIcon } from '@chakra-ui/icons';
 import PropTypes from 'prop-types';
 import styles from './BoxInfo.module.css';
 import { FYABackend } from '../../../common/utils';
+import DeleteBoxModal from '../DeleteBoxModal/DeleteBoxModal';
 
-const BoxInfo = ({ selectedBox, setSelectedBox }) => {
+const BoxInfo = ({
+  selectedBox,
+  setSelectedBox,
+  selectedZipCode,
+  selectedCountry,
+  setSelectedZipCode,
+  setSelectedCountry,
+  zipCodeData,
+  setZipCodeData,
+}) => {
   const [boxHistory, setBoxHistory] = useState([]);
+
+  const {
+    isOpen: isOpenDeleteBoxModal,
+    onOpen: onOpenDeleteBoxModal,
+    onClose: onCloseDeleteBoxModal,
+  } = useDisclosure();
+
   useEffect(async () => {
     const response = await FYABackend.get(`/boxHistory/history/${selectedBox.box_id}`);
     setBoxHistory(response.data);
   }, []);
+  // Deletes the currently selected box in both Anchor_Box and Box_History
+  const deleteBox = async () => {
```

Comment:
I'm getting a bug where if I delete a box from the pin at 92130, then the 92697 pin disappears. The pin comes back if I refresh the page. Could you take a look into why this may be happening?